### ISSUE TITLE
[zod-openapi] Problem with strict type checking of return values

### ISSUE THREAD

**Issue by WiktorKryzia** [honojs/middleware#374]
Strict type checking was added in version **0.9.1** in such a way that if response type is declared, the handler of a given route can only return a response of the `TypedResponse` type, and of the methods available in Hono Context, only the `.json()` (and deprecated `.jsonT()`) method returns such a type, which forces that only JSON can be returned from this endpoint.

It's about this piece of code:
```ts
// If response type is defined, only TypedResponse is allowed.
  R extends {
    responses: {
      [statusCode: string]: {
        content: {
          [mediaType: string]: ZodMediaTypeObject
        }
      }
    }
  }
    ? HandlerTypedResponse<OutputType<R>>
    : HandlerAllResponse<OutputType<R>>
```

As a result, if I want to have a route that returns text or e.g. YAML, TS will report an error because the methods `Context.text()`, `Context.body()` etc. return the `Response` type, but `TypedResponse` is expected.

**Comment by yusukebe** [comment]
Hi @WiktorKryzia 

Indeed, only `TypeResponse` is now allowed. But I think this is fine.

Since `c.text()` and `c.body()` cannot return a type, they cannot be defined in the `ZodMediaTypeObject` schema. So, if you want to return a response with `c.text()` or `c.body()`, write like this:

```ts
const route = createRoute({
  method: 'get',
  path: '/hello',
  responses: {
    200: {
      description: 'Say hello'
    }
  }
})

app.openapi(route, (c) => {
  return c.text('hello')
})
```

**Comment by yusukebe** [comment]
@WiktorKryzia 

I understand what you want to do. Thanks for the explanation.

Certainly it would be better to be able to specify content other than JSON. However, since shema supports only JSON, it would be better to make schema optional or undefined for `text/plain`.

```ts
const route = createRoute({
  method: 'get',
  path: '/info',
  responses: {
    200: {
      description: 'OK',
      content: {
        'text/plain': {
          example: 'API v1.0.0',
          schema: undefined
        }
      }
    }
  }
})
```

This feature can be added.

**Comment by WiktorKryzia** [comment]
Yes exactly, I think it would be really nice to add such an option.

**Comment by yusukebe** [comment]
@WiktorKryzia 

Thanks! I'll work on it later.

**Comment by marvinruder** [comment]
Related: #641 

@yusukebe Allowing `schema: undefined` seems to be a nice solution. Will this be implemented soon?

**Comment by abielzulio** [comment]
any updates?
